// @axelar-network/axelarjs-sdk/dist/src/services/SocketService.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SocketService = void 0;
const socket_io_client_1 = require("socket.io-client");
class SocketService {
    constructor(resourceUrl, testMode = false) {
        this.testMode = false;
        this.resourceUrl = resourceUrl;
        this.testMode = testMode;
    }
    createSocket() {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.testMode) {
                this.socket = (0, socket_io_client_1.io)(this.resourceUrl, {
                    reconnectionDelay: 0,
                    forceNew: true,
                    transports: ["websocket"],
                    extraHeaders: {
                        "User-Agent": "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0",
                    },
                });
            }
            else {
                this.socket = (0, socket_io_client_1.io)(this.resourceUrl, {
                    transports: ["websocket"],
                    reconnectionDelayMax: 10000,
                    extraHeaders: {
                        "User-Agent": "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0",
                    },
                });
            }
            return new Promise((resolve) => {
                this.socket.on("connect", () => {
                    resolve(true);
                });
            });
        });
    }
    joinRoomAndWaitForEvent(roomId) {
        return new Promise((resolve) => __awaiter(this, void 0, void 0, function* () {
            yield this.createSocket();
            const ms = 1.8e6; //30 minutes
            const timeout = setTimeout(() => {
                this.disconnect();
            }, ms);
            this.socket.emit("room:join", roomId, () => {
                this.socket.on("bridge-event", (data) => {
                    clearTimeout(timeout);
                    resolve(data);
                    this.disconnect();
                });
            });
        }));
    }
    disconnect() {
        this.socket.disconnect();
    }
}
exports.SocketService = SocketService;
