// @axelar-network/axelarjs-sdk/dist/src/libs/test/stubs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.newRoomIdStub = exports.linkEventStub = exports.depositAddressPayloadStub = exports.roomIdStub = exports.apiErrorStub = exports.otcStub = exports.terraAddressStub = exports.ethAddressStub = exports.uuidStub = void 0;
var uuidStub = function () { return "83462f97-63cf-4205-a659-6f54bec623f6"; };
exports.uuidStub = uuidStub;
var ethAddressStub = function () {
    return "0xF16DfB26e1FEc993E085092563ECFAEaDa7eD7fD";
};
exports.ethAddressStub = ethAddressStub;
var terraAddressStub = function () {
    return "terra1qem4njhac8azalrav7shvp06myhqldpmkk3p0t";
};
exports.terraAddressStub = terraAddressStub;
var otcStub = function () { return ({
    otc: "hr64_XnjNE",
    validationMsg: "Verify I'm a real user with this one-time-code: hr64_XnjNE (This will not cost any fees)",
}); };
exports.otcStub = otcStub;
var apiErrorStub = function () { return ({
    message: "AxelarJS-SDK uncaught post error",
    uncaught: true,
    fullMessage: "",
}); };
exports.apiErrorStub = apiErrorStub;
var roomIdStub = function () { return ({
    roomId: '{"assetCommonKey":"uusd","destinationAddress":"0xF16DfB26e1FEc993E085092563ECFAEaDa7eD7fD","destinationChainIdentifier":"avalanche","sourceModule":"axelarnet","type":"link"}',
}); };
exports.roomIdStub = roomIdStub;
var depositAddressPayloadStub = function () { return ({
    fromChain: "terra",
    toChain: "avalanche",
    asset: "uusd",
    publicAddress: (0, exports.ethAddressStub)(),
    destinationAddress: (0, exports.ethAddressStub)(),
    signature: "0x",
}); };
exports.depositAddressPayloadStub = depositAddressPayloadStub;
var linkEventStub = function () { return ({
    Type: "link",
    Attributes: {
        asset: "uusd",
        depositAddress: "axelar1ncj6rh3fxzx6pz35gjc6fva534vqmpsx3pwj8s8uyrqrthrch4asgu5l0p",
        destinationAddress: "0xF16DfB26e1FEc993E085092563ECFAEaDa7eD7fD",
        destinationChain: "Avalanche",
        module: "axelarnet",
        sourceChain: "Axelarnet",
    },
    Height: 1502111,
    newRoomId: '{"depositAddress":"axelar1ncj6rh3fxzx6pz35gjc6fva534vqmpsx3pwj8s8uyrqrthrch4asgu5l0p","sourceModule":"axelarnet","type":"deposit-confirmation"}',
}); };
exports.linkEventStub = linkEventStub;
var newRoomIdStub = function () {
    return '{"depositAddress":"axelar1ncj6rh3fxzx6pz35gjc6fva534vqmpsx3pwj8s8uyrqrthrch4asgu5l0p","sourceModule":"axelarnet","type":"deposit-confirmation"}';
};
exports.newRoomIdStub = newRoomIdStub;
