// @axelar-network/axelarjs-sdk/dist/src/chains/Terra/index.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var Axelar_1 = __importDefault(require("../Axelar"));
var terra_js_1 = require("@terra-money/terra.js");
var Terra = /** @class */ (function (_super) {
    __extends(Terra, _super);
    function Terra() {
        var _this = _super.call(this) || this;
        _this.chainInfo = {
            chainSymbol: "Terra",
            chainName: "Terra",
            estimatedWaitTime: 5,
            fullySupported: true,
            assets: [],
            txFeeInPercent: 0.1,
            module: "axelarnet",
            chainIdentifier: {
                devnet: "terra-2",
                testnet: "terra-2",
                mainnet: "terra-2",
            },
        };
        _this.validateAddress = function (address) { return terra_js_1.AccAddress.validate(address); };
        return _this;
    }
    return Terra;
}(Axelar_1.default));
exports.default = Terra;
