// @axelar-network/axelarjs-sdk/dist/src/chains/Terra/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Axelar_1 = __importDefault(require("../Axelar"));
const bech32_1 = require("bech32");
class Terra extends Axelar_1.default {
    constructor() {
        super();
        this.chainInfo = {
            chainSymbol: "Terra",
            chainName: "Terra",
            estimatedWaitTime: 5,
            fullySupported: true,
            assets: [],
            txFeeInPercent: 0.1,
            module: "axelarnet",
            chainIdentifier: {
                devnet: "terra-2",
                testnet: "terra-2",
                mainnet: "terra-2",
            },
        };
        this.validateAddress = (address) => checkPrefixAndLength("terra", address, 44) || checkPrefixAndLength("terra", address, 64);
    }
}
exports.default = Terra;
function checkPrefixAndLength(prefix, data, length) {
    try {
        const vals = bech32_1.bech32.decode(data);
        return vals.prefix === prefix && data.length == length;
    }
    catch (e) {
        return false;
    }
}
