// @axelar-network/axelarjs-sdk/dist/src/libs/GatewayTx.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GatewayTx = void 0;
var GatewayTx = /** @class */ (function () {
    function GatewayTx(unsignedTx, provider) {
        this.txRequest = {
            to: unsignedTx.to,
            data: unsignedTx.data,
            value: unsignedTx.value,
        };
        this.provider = provider;
    }
    GatewayTx.prototype.send = function (signer, txOption) {
        var txRequest = __assign(__assign({}, this.txRequest), txOption);
        return signer.connect(this.provider).sendTransaction(txRequest);
    };
    GatewayTx.prototype.estimateGas = function () {
        return this.provider.estimateGas(this.txRequest);
    };
    return GatewayTx;
}());
exports.GatewayTx = GatewayTx;
