// @axelar-network/axelarjs-sdk/dist/src/chains/Injective/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Axelar_1 = __importDefault(require("../Axelar"));
const bech32_1 = require("bech32");
class Injective extends Axelar_1.default {
    constructor() {
        super();
        this.chainInfo = {
            chainSymbol: "INJ",
            chainName: "Injective",
            estimatedWaitTime: 5,
            fullySupported: true,
            assets: [],
            txFeeInPercent: 0.1,
            module: "axelarnet",
            chainIdentifier: {
                devnet: "injective",
                testnet: "injective",
                mainnet: "injective",
            },
        };
        this.validateAddress = (address) => {
            if (!address)
                return false;
            try {
                return bech32_1.bech32.decode(address).prefix === this.chainInfo.chainSymbol.toLowerCase();
            }
            catch (e) {
                return false;
            }
        };
    }
}
exports.default = Injective;
