// @axelar-network/axelarjs-sdk/dist/src/chains/Crescent/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const bech32_1 = require("bech32");
const Axelar_1 = __importDefault(require("../Axelar"));
class Crescent extends Axelar_1.default {
    constructor() {
        super();
        this.chainInfo = {
            chainSymbol: "CRE",
            chainName: "Crescent",
            estimatedWaitTime: 5,
            fullySupported: true,
            assets: [],
            txFeeInPercent: 0.1,
            module: "axelarnet",
            chainIdentifier: {
                devnet: "crescent",
                testnet: "crescent",
                mainnet: "crescent",
            },
        };
        this.validateAddress = (address) => {
            if (!address)
                return false;
            try {
                return bech32_1.bech32.decode(address).prefix === "cre";
            }
            catch (e) {
                return false;
            }
        };
    }
}
exports.default = Crescent;
