// @axelar-network/axelarjs-sdk/dist/src/chains/Fantom/index.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var Ethereum_1 = __importDefault(require("../Ethereum"));
var Fantom = /** @class */ (function (_super) {
    __extends(Fantom, _super);
    function Fantom() {
        var _this = _super.call(this) || this;
        _this.chainInfo = {
            chainSymbol: "FTM",
            chainName: "Fantom",
            estimatedWaitTime: 5,
            fullySupported: true,
            assets: [],
            txFeeInPercent: 0.1,
            module: "evm",
            confirmLevel: 6,
            chainIdentifier: {
                devnet: "fantom",
                testnet: "fantom",
                mainnet: "fantom",
            },
        };
        return _this;
    }
    return Fantom;
}(Ethereum_1.default));
exports.default = Fantom;
