// @axelar-network/axelarjs-sdk/dist/src/chains/Axelar/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var bech32_1 = require("bech32");
var Axelar = /** @class */ (function () {
    function Axelar() {
        var _this = this;
        this.chainInfo = {
            chainSymbol: "AXL",
            chainName: "Axelar",
            estimatedWaitTime: 5,
            fullySupported: false,
            assets: [],
            txFeeInPercent: 0.1,
            module: "axelarnet",
            chainIdentifier: {
                devnet: "axelarnet",
                testnet: "axelarnet",
                mainnet: "axelarnet",
            },
        };
        this.validateAddress = function (address) {
            if (!address)
                return false;
            try {
                return (bech32_1.bech32.decode(address).prefix === _this.chainInfo.chainName.toLowerCase());
            }
            catch (e) {
                return false;
            }
        };
    }
    return Axelar;
}());
exports.default = Axelar;
