// @axelar-network/axelarjs-sdk/dist/src/constants/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getConfigs = void 0;
var configsMap = {};
var localConfigs = {
    resourceUrl: "http://localhost:4000",
    axelarRpcUrl: "https://axelartest-rpc.quickapi.com",
    axelarLcdUrl: "https://axelartest-lcd.quickapi.com",
    axelarCachingServiceUrl: "https://testnet.api.gmp.axelarscan.io"
};
var devnetConfigs = {
    resourceUrl: "https://nest-server-devnet.axelar.dev",
    axelarRpcUrl: "",
    axelarLcdUrl: "",
    axelarCachingServiceUrl: "https://devnet.api.gmp.axelarscan.io"
};
var testnetConfigs = {
    resourceUrl: "https://nest-server-testnet.axelar.dev",
    axelarRpcUrl: "https://axelartest-rpc.quickapi.com",
    axelarLcdUrl: "https://axelartest-lcd.quickapi.com",
    axelarCachingServiceUrl: "https://testnet.api.gmp.axelarscan.io"
};
var mainnetConfigs = {
    resourceUrl: "https://nest-server-mainnet.axelar.dev",
    axelarRpcUrl: "https://axelar-rpc.quickapi.com",
    axelarLcdUrl: "https://axelar-lcd.quickapi.com",
    axelarCachingServiceUrl: "https://mainnet.api.gmp.axelarscan.io"
};
configsMap["local"] = localConfigs;
configsMap["devnet"] = devnetConfigs;
configsMap["testnet"] = testnetConfigs;
configsMap["mainnet"] = mainnetConfigs;
var configToUse;
var getConfigs = function (environment) {
    if (!configToUse) {
        if (!configsMap[environment])
            throw new Error("config environment does not exist");
        configToUse = configsMap[environment];
    }
    return configToUse;
};
exports.getConfigs = getConfigs;
